app/elements/pw-shell.html

```diff
@@ -365,9 +366,27 @@
       return data.version+'/'+data.section;
     },
 
+    _handleError: function(event) {
+      window.showToast('This page was not cached for offline use :(');
+    },
+
     _handleResponse: function(event) {
-      // Don't re-load the same thing you're looking at.
-      if (this._currentPageURL !== this._sectionRoute.path) {
+      // We only want to switch out the app-shell content if:
+      // - we are not refreshing the navigate fallback page (the server is already
+      // serving the correct page, and we don't need to load it twice)
+      // - this is a page to page transition (i.e. not an initial navigation) or
+      // - this is an initial navigation _and_ it was served by a Service Worker,
+      // since that is going to just give us the shell.html and we need to
+      // actually swap its content out.
+      var hasServiceWorker = navigator.serviceWorker && navigator.serviceWorker.controller;
+      var isNavigateFallback = this._isInitialNavigation && this._sectionRoute.path === '/';
```

Comment:
After I have the SW installed, if I reload on a URL that isn't cached (say https://service-worker-dot-polymer-project.appspot.com/2.0/docs/devguide/properties, but really any page that's not `/`). I get `/shell` and `<pw-shell>._isInitialNavigation` is true by the time the page is done loading (`_handleResponse` is never called since you're offline and won't get a response for an uncached page). Now if I click on the logo/homepage link, the navigation fails because`isNavigateFallback` (this line) will be true.

One way to fix this is to do `this._isInitialNavigation = false;` in `_handleError` as well. The more robust way IMO is to hopefully get what I mentioned to @jeffposnick in the gulpfile comment, which is don't serve `/shell` if the content is cached so that If we do get `/shell` then we know that there's no content to swap, and future navigation wouldn't need to go through this check (in fact, pw-shell would behave the same with or without a service worker, which from a web dev point a view should be the case since the SW only acts as a network proxy).